src/radical/pilot/agent/lm/yarn.py

```diff
@@ -160,7 +215,12 @@ def config_yarn_site():
             # Solution to find Java's home folder:
             # http://stackoverflow.com/questions/1117398/java-home-directory
 
-            jpos = subprocess.check_output(['readlink','-f', '/usr/bin/java']).split('bin')
+            java = subprocess.check_output(['which','java'])
```

Comment:
there is this:  https://github.com/radical-cybertools/radical.utils/blob/devel/src/radical/utils/which.py#L9